models/pull.go

```diff
@@ -232,15 +248,36 @@ func (pr *PullRequest) CanAutoMerge() bool {
 	return pr.Status == PullRequestStatusMergeable
 }
 
+// MergeStyle represents the approach to merge commits into base branch.
+type MergeStyle string
+
+const (
+	// MergeStyleRegular create merge commit
+	MergeStyleRegular MergeStyle = "merge"
+	// MergeStyleRebase rebase before merging
+	MergeStyleRebase MergeStyle = "rebase"
+	// MergeStyleSquash squash commits into single commit before merging
+	MergeStyleSquash MergeStyle = "squash"
+)
+
 // Merge merges pull request to base repository.
 // FIXME: add repoWorkingPull make sure two merges does not happen at same time.
-func (pr *PullRequest) Merge(doer *User, baseGitRepo *git.Repository) (err error) {
+func (pr *PullRequest) Merge(doer *User, baseGitRepo *git.Repository, mergeStyle MergeStyle, message string) (err error) {
 	if err = pr.GetHeadRepo(); err != nil {
 		return fmt.Errorf("GetHeadRepo: %v", err)
 	} else if err = pr.GetBaseRepo(); err != nil {
 		return fmt.Errorf("GetBaseRepo: %v", err)
 	}
 
+	pullsConfig := pr.BaseRepo.MustGetUnit(UnitTypePullRequests).PullRequestsConfig()
+
+	// Check if merge style is correct and allowed
+	if mergeStyle == MergeStyleRegular && !pullsConfig.AllowMerge ||
```

Comment:
This big boolean expression appears multiple times, maybe move to a method on `PullRequestsConfig`?